fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/FieldValue.java

```diff
@@ -174,6 +176,14 @@ private static Object unwrapPrimitive(@Nonnull Type type, @Nullable Object field
             return returnList;
         } else if (type.getTypeCode() == Type.TypeCode.VERSION) {
             return FDBRecordVersion.fromBytes(((ByteString)fieldValue).toByteArray(), false);
+        } else if (type.isUuid()) {
+            if (fieldValue instanceof UUID) {
```

Comment:
This cannot happen I think. Can you remove this?

Reply:
Make sense!

Reply:
The UUID seems to have been "unwrapped" before itself. On the contrary, when we reach this point Type UUID always has a value of Java UUID. 

This is a bit complex in the way how unwrapping is done. We have Nullalbe arrays and Versions that unwrap here, whereas TupleFields (UUID, Nullable primitives) are already unwrapped downstream.

Reply:
Ah crap, this is considered a tuple field.